middleware/database_session.py

```diff
@@ -1,143 +1,133 @@
-
-    async def dispatch(self, request: Request, call_next):
-        info = self._prepare_info(request)
-        await self.set_body(request)
-
-        request.state.session_token = general.get_ctx_token()
-        info.context = {"request": request}
-        request.state.info = info
-        request.state.parsed = {}
-
-        if request.url.hostname != "localhost" or request.url.port != 7051:
-            access_response = self._check_access(request, info)
-            if access_response is not None:
-                general.remove_and_refresh_session(request.state.session_token)
-                return access_response
-        # sets states so called before actual request handling
-        log_request = auth_manager.extract_state_info(request, "log_request")
-        try:
-            response = await call_next(request)
-        finally:
-            if log_request:
-                # after call next so the path_params are mapped
-                await self._log_request(request)
-            general.remove_and_refresh_session(request.state.session_token)
-        return response
+async def handle_db_session(request: Request, call_next):
+    info = _prepare_info(request)
 
-    def _check_access(self, request, info):
-        try:
-            auth_manager.check_demo_access(info)
-        except NotAllowedInDemoError:
-            general.remove_and_refresh_session(request.state.session_token)
-            return JSONResponse(
-                status_code=401,
-                content={"message": "Unauthorized access"},
-            )
-        except GraphQLError as e:
-            general.remove_and_refresh_session(request.state.session_token)
-            return JSONResponse(
-                status_code=400,
-                content={"message": e.message},
-            )
-        except ValueError as e:
-            general.remove_and_refresh_session(request.state.session_token)
-            return JSONResponse(
-                status_code=400,
-                content={"message": str(e)},
-            )
-        except Exception:
+    request.state.session_token = general.get_ctx_token()
+    info.context = {"request": request}
+    request.state.info = info
+    request.state.parsed = {}
+
+    if request.url.hostname != "localhost" or request.url.port != 7051:
+        access_response = _check_access(request, info)
+        if access_response is not None:
             general.remove_and_refresh_session(request.state.session_token)
-            return JSONResponse(
-                status_code=500,
-                content={"message": "Internal server error"},
-            )
+            return access_response
 
-        return None
+    log_request = auth_manager.extract_state_info(request, "log_request")
+    length = request.headers.get("content-length")
 
-    def _prepare_info(self, request):
-        field_name = None
-        parent_type = None
+    if log_request and length and int(length) > 0:
+        await request.body()
```

Comment:
`await log_storage.set_request_data(request)` (the function used in gates / cognition) should be used and adapted. The await request.body() doesn't do much only awaiting. The value also needs to be stored in a state. 

From gates:

```
async def set_request_data(request: Request) -> bytes:
    data = None
    length = request.headers.get("content-length")
    if length and int(length) > 0:
        if request.headers.get("Content-Type") == "application/json":
            data = await request.json()
        else:
            data = await request.body()
    request.state.data = data

```

and then inside the log_request function the data is used where it's available

```
    data = None
    if hasattr(request.state, "data"):
        data = request.state.data
```